### PR TITLE
[004] Update GetDelivery Employee to return all information

### PR DESCRIPTION
**Pull request by TaylorJCrawford**

### CODE REVIEW

File: src/main/java/org/example/api/DeliveryEmployeeService.java [hunk 1]
```diff
@@ -49,4 +52,16 @@ public DeliveryEmployee getDeliveryEmployeeById(int id) throws DoesNotExistExcep
...
     }
+
+    public List<DeliveryEmployee> getDeliveryEmployees() throws DoesNotExistException, SQLException {
+
+        // Call Get ALL DeliveryEmplyoees
+        List<DeliveryEmployee> deliveryEmployeeList = dao.getDeliveryEmployees();
+
+        if (deliveryEmployeeList.size() == 0) {
+            throw new DoesNotExistException();
+        }
```
Comment (shaunganley) [inline src/main/java/org/example/api/DeliveryEmployeeService.java]: you don't need this logic, just return an empty list

File: src/main/java/org/example/resources/DeliveryEmployeeController.java [hunk 2]
```diff
@@ -25,8 +26,12 @@ public class DeliveryEmployeeController {
     @Path("/employee/delivery")
     @Produces(MediaType.APPLICATION_JSON)
     public Response getAllEmployees(){
-        DeliveryEmployeeDAO employeeDAO = new DeliveryEmployeeDAO();
-        return Response.ok().entity(employeeDAO.getDeliveryEmployeesIds()).build();
+        try {
+            DeliveryEmployeeDAO employeeDAO = new DeliveryEmployeeDAO();
```
Comment (shaunganley) [inline src/main/java/org/example/resources/DeliveryEmployeeController.java]: move to the top of class

File: src/main/java/org/example/resources/DeliveryEmployeeController.java [hunk 3]
```diff
@@ -25,8 +26,12 @@ public class DeliveryEmployeeController {
...
+            return Response.ok().entity(employeeDAO.getDeliveryEmployees()).build();
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
```
Comment (shaunganley) [inline src/main/java/org/example/resources/DeliveryEmployeeController.java]: return 500 response

**Review by shaunganley — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/api/DeliveryEmployeeService.java (hunk 1 above)
Comment (miagreen23) [reply src/main/java/org/example/api/DeliveryEmployeeService.java]: Amended

File: src/main/java/org/example/resources/DeliveryEmployeeController.java (hunk 2 above)
Comment (miagreen23) [reply src/main/java/org/example/resources/DeliveryEmployeeController.java]: Amended (we think)

File: src/main/java/org/example/resources/DeliveryEmployeeController.java (hunk 3 above)
Comment (miagreen23) [reply src/main/java/org/example/resources/DeliveryEmployeeController.java]: Amended

File: src/main/java/org/example/api/DeliveryEmployeeService.java [hunk 4]
```diff
@@ -75,4 +76,16 @@ public DeliveryEmployee getDeliveryEmployeeById(int id) throws DoesNotExistExcep
...
     }
+
+    public List<DeliveryEmployee> getDeliveryEmployees() throws DoesNotExistException, SQLException {
+
+        // Call Get ALL DeliveryEmplyoees
+        List<DeliveryEmployee> deliveryEmployeeList = dao.getDeliveryEmployees();
+
+        if (deliveryEmployeeList.size() == 0) {
+            return deliveryEmployeeList;
+        }
```
Comment (shaunganley) [inline src/main/java/org/example/api/DeliveryEmployeeService.java]: you can remove this as it just does the same thing as the line below

**Review by shaunganley — APPROVED** [review]

File: src/main/java/org/example/api/DeliveryEmployeeService.java (hunk 4 above)
Comment (miagreen23) [reply src/main/java/org/example/api/DeliveryEmployeeService.java]: Removed

**Pull request merged**